Autoreply to Envelope FROM

This patch is from the mailing list, and allows auto-reply
to the value of the Envelope 'FROM'.

Index: cyrus-imapd-2.3.12rc1/sieve/script.c
===================================================================
--- cyrus-imapd-2.3.12rc1.orig/sieve/script.c	2008-04-09 07:43:52.000000000 -0400
+++ cyrus-imapd-2.3.12rc1/sieve/script.c	2008-04-09 07:48:49.000000000 -0400
@@ -449,6 +449,10 @@ static int send_notify_callback(sieve_in
     nc.options = notify->options ? notify->options : NULL;
     nc.priority = notify->priority;
 
+    if(nc.options && !strcmp(nc.method,"mailto"))
+      if(!strcmp("$env-from$",*nc.options))
+        interp->getenvelope(message_context, "From", &nc.options);
+
     build_notify_message(interp, body_cache, notify->message, message_context, 
 			 &out_msg, &out_msglen);
 
